### DISCUSSION TITLE
FATAL:  database "teslamate" does not exist.  Clock issue or?

### DISCUSSION THREAD

**Discussion by JonMurphy** [teslamate-org/teslamate#2053] [Getting started] [answered]
Hello all!  

**EDIT**: Forgot to add something important - this is using the Docker install:
https://docs.teslamate.org/docs/installation/docker

I cannot get a new installation to run correctly.  Going to `http://192.168.60.235:4000` does not connect.  Grafana at `http://192.168.60.235:3000` seems to be working.

In Grafana I see a bright red error:
```
Templating [car_id]
Error updating options: pq: database "teslamate" does not exist
```

And when I enter: `docker-compose down` and then `docker-compose up` I see:
```
teslamate_1  | 2021-10-12 20:49:12.109 [error] Could not create schema migrations table. This error usually happens due to the following:
```
-and-
```
database_1   | 1970-01-01 00:00:00.000 UTC [38] FATAL:  database "teslamate" does not exist
teslamate_1  | 2021-10-12 20:49:45.133 [error] Postgrex.Protocol (#PID<0.165.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "teslamate" does not exist
```
-and-
```
teslamate_teslamate_1 exited with code 1
```
-and-
```
pi@teslamate:~/teslamate $ docker-compose ps
        Name                       Command                 State                       Ports                  
--------------------------------------------------------------------------------------------------------------
teslamate_database_1    docker-entrypoint.sh postgres    Up           5432/tcp                                
teslamate_grafana_1     /run.sh                          Up           0.0.0.0:3000->3000/tcp,:::3000->3000/tcp
teslamate_mosquitto_1   /docker-entrypoint.sh mosq ...   Up           0.0.0.0:1883->1883/tcp,:::1883->1883/tcp
teslamate_teslamate_1   tini -- /bin/sh /entrypoin ...   Restarting                                           
```

So all the good hints seem to be in the log but I am not understanding what I need to do to fix them.  I've dug the the Teslamate GitHub **Issues** area and thru the **Discussions** (still going thru this area) and I find similar issues but I am missing the fix.

I tried building (rebuilding?) the teslamate_teslamate_1  with `docker-compose up -d --no-deps --build <service_name>` and that did not help.

I noticed the database clock is wrong `1970-01-01 00:00:00.000 UTC `.  Does that matter?

I set up TeslaMate late year and all has worked great.  But I did a recent update of teslamate, ran out of SD space and bad happened.  So I started with a new SD card and new raspberry pi Rasbian build.

Can someone lend a hand and point me in the right direction?

Jon

[docker-compose up - Terminal Log copy.txt](https://github.com/adriankumpf/teslamate/files/7333196/docker-compose.up.-.Terminal.Log.copy.txt)

**Comment by pjkeller** [comment]
There are two of us on the TMC (https://teslamotorsclub.com/tmc/threads/teslamate-megathread.191537/page-86) web site that are experiencing the same failure but so far no solutions, just empathy.

**Reply by JonMurphy** [reply]
Thank you for letting me know!  I had been reading that L-O-N-G thread but I am only in the 60's out of 85 pages!  I'll skip ahead!

**Comment by JonMurphy** [comment]
Hello @adriankumpf - I am hoping you can offer a hint or three.  

There are a few monitoring via https://teslamotorsclub.com also!

**Comment by JonMurphy** [comment, marked as answer]
I changed my docker-compose.yml from `image: postgres:13` to `image: postgres:12` and all of the above errors disappeared!

a HUGE "thank you" to @dongbum for posting a message on this page:
https://github.com/adriankumpf/teslamate/commit/0831124f08b46b58fba136fb7d90144930b2afaa#diff-f552d1b58d6a83752392d8c141a978ae739214bd842892f4b8d8810b56f1ae11

I posted here also:
https://teslamotorsclub.com/tmc/posts/6044594/

tl;dr - This was all installed on an RPi4B with 1GB RAM and 16 GB sd card.  Running the current Buster build plus various apt updates.  Raspberry Pi OS Lite, Release date: May 7th 2021, Kernel version: 5.10

**Comment by osvedman** [comment]
I'm trying to set up TeslaMate on an RPi4B for the first time and I'm still having this issue even after changing Postgres from 13 to 12 in the docker-compose.yml and performing the same commands as you posted on https://teslamotorsclub.com/tmc/posts/6044594/

**Reply by JonMurphy** [reply]
I found doing a simple change from 13 to 12 didn't do any good.  I had to "cleanse" anything Docker-ish. 

If you've done that AND If you're having the same errors as I was experiencing above then I'd suggest starting from scratch.  Start with a blank SD card, add Raspian Buster and then walk thru the Docker steps on Teslamate.  Maybe that will help.  

Honestly I am guessing here.  Just so you know I am not an expert with Teslamate.  I just stumbled across an answer and got lucky.  I'm sure @adriankumpf and others can give you better advice.  

Hope this helps!

**Reply by JonMurphy** [reply]
TJ63 - Many more people will see your post if you start new (go to **Discussions** > **New discussion**).   After you've started the new Discussion then come back here and delete the above post.  Locate the 3 dots on the far right of your avatar and name, then click **Delete**.  Otherwise with duplicate post things will become confusing!

FYI - the video issue is different.  It's **not** related to Teslamate.  But can be solved!

**Reply by tsjames6** [reply]
Thanks Jon. Sorted out now.
